java/algorithms/Trie-FunSat01/Main.java

```diff
@@ -0,0 +1,286 @@
+   * @return all keys in the sybol table as an <tt>Iterable</tt>
+   */
+
+
+  /**
+   * Returns all of the keys in the set that start with <tt>prefix</tt>.
+   * @param prefix the prefix
+   * @return all of the keys in the set that start with <tt>prefix</tt>,
+   *     as an iterable
+   */
+
+
+  /**
+   * Returns all of the keys in the symbol table that match <tt>pattern</tt>,
+   * where . symbol is treated as a wildcard character.
+   * @param pattern the pattern
+   * @return all of the keys in the symbol table that match <tt>pattern</tt>,
+   *     as an iterable, where . is treated as a wildcard character.
+
+
+
+
+   /**
+   * Returns the string in the symbol table that is the longest prefix of <tt>query</tt>,
+   * or <tt>null</tt>, if no such string.
+   * @param query the query string
+   * @return the string in the symbol table that is the longest prefix of <tt>query</tt>,
+   *     or <tt>null</tt> if no such string
+   * @throws NullPointerException if <tt>query</tt> is <tt>null</tt>
+   */
+  public CharArray longestPrefixOf(CharArray query) {
+    int length = longestPrefixOf(root, query, 0, -1);
+    if (length == -1) return null;
+    else return query.substring(0, length);
+  }
+
+  // returns the length of the longest string key in the subtrie
+  // rooted at x that is a prefix of the query string,
+  // assuming the first d character match and we have already
+  // found a prefix match of given length (-1 if no such match)
+  private int longestPrefixOf(Node x, CharArray query, int d, int length) {
+    if (x == null) return length;
+    if (x.val != -1) length = d;
+    if (d == query.length()) return length;
+    char c = query.get(d);
+    return longestPrefixOf(x.next[c], query, d+1, length);
+  }
+
+  /**
+   * Removes the key from the set if the key is present.
+   * @param key the key
+   * @throws NullPointerException if <tt>key</tt> is <tt>null</tt>
+   */
+  public void delete(CharArray key) {
+    root = delete(root, key, 0);
+  }
+
+  private Node delete(Node x, CharArray key, int d) {
+    if (x == null) return null;
+    if (d == key.length()) {
+      if (x.val != -1) N--;
+      x.val = -1;
+    }
+    else {
+      char c = key.get(d);
+      x.next[c] = delete(x.next[c], key, d+1);
+    }
+
+    // remove subtrie rooted at x if it is completely empty
+    if (x.val != -1) return x;
+    for (int c = 0; c < R; c++)
+      if (x.next[c] != null)
+        return x;
+    return null;
+  }
+
+
+  public static void main(String[] args) {
+    Main st = new Main();
+    CharArray test = new CharArray(8);
+
+    try{
+      test.set(0, 'S');
+      test.set(1, 'V');
+      test.set(2, 'C');
+      test.set(3, 'O');
+      test.set(4, 'M');
+      test.set(5, 'P');
+      test.set(6, '2');
+      test.set(7, '0');
+
+      int val = Verifier.nondetInt();
+      st.put(test, val);
+      assert (st.get(test) == val);
+    } catch(Exception e) { assert false; }
```

Comment:
After reading through the code, I am not sure, whether this catch is ever supposed to be triggered.
Otherwise, might the try-catch block be removed?

Reply:
Yes, I've removed it for the benchmarks checking functional safety. For the benchmarks checking memory safety, I guess the answer will depend on the conclusion of [this conversaton](https://github.com/sosy-lab/sv-benchmarks/pull/823#discussion_r338071487).